plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d

```diff
@@ -678,7 +744,15 @@ void generateFile(ref Database db, ref FileCtx ctx) @trusted {
             line.setAttribute("id", format("%s-%s", "loc", lastLoc.line + i + 1))
                 .addClass("loc").addChild("span", format("%s:",
                         lastLoc.line + i + 1)).addClass("line_nr");
-
+            
+            if (coverageData.lineExists(lastLoc.line + i + 1)) {
```

Comment:
Line 748-749 is two lookups. I think you can reduce it by doing one lookup and then check the variable.

```d
if (auto v = (lastLoc.line + i + 1) in coverageData.lineList) {
  if (*v) line.addClass("loc_covered");
  else line.addClass("loc_noncovered");
}
```

"in" returns a pointer to the element value if key exist, otherwise `null`.